core/src/main/java/org/jdbi/v3/core/statement/Cleanable.java

```diff
@@ -27,4 +27,23 @@
 public interface Cleanable extends AutoCloseable {
     @Override
     void close() throws SQLException;
+
+    /*
+     * TODO - This is *NOT* the final location for these methods. But I needed something that is public and somewhat related to cleaning up.
+     */
+
+    static void cleanUp(BaseStatement<?> statement) {
+        if (statement != null) {
+            statement.getContext().close();
+        }
+    }
+
+    static void cleanUpForException(SQLException e, BaseStatement<?> statement) {
+
+        try {
+            cleanUp(statement);
+        } catch (Exception e1) {
+            e.addSuppressed(e1.getCause());
```

Comment:
This may not have a cause, in which case it will throw a NPE as adding a `null` surpressed exception is not allowed.

Reply:
good catch.

Reply:
I was noodling around with catching just the CloseException (which comes out of the StatementContext close) but then changed it back to the original "catch Exception" but did not change the getCause back. I now treat these exceptions differently (CloseException gets unwound, everything else is added "as is").